weave-js/src/components/Panel2/useAssetFromArtifact.ts

```diff
@@ -30,7 +30,11 @@ export const useAssetURLFromArtifact = <
     if (!nodeValueQuery.loading && nodeValueQuery.result != null) {
       return opArtifactVersionFile({
         artifactVersion: opAssetArtifactVersion({asset: inputNode}),
-        path: constString(nodeValueQuery.result.path),
+        path: constString(
+          typeof nodeValueQuery.result === `string`
+            ? nodeValueQuery.result
+            : nodeValueQuery.result.path
+        ),
```

Comment:
Fix 2

Reply:
this feels like a smell. under what conditions is the result itself a string vs dict?

Reply:
I agree that it's a smell, but it was necessary in the initial fix attempt, when `_process_run_dict_item` directly returned `_filesystem_runfiles_from_run_path` just like for `table-file`s.

Now that we're using the `XXXArtifactFileRef` path instead, it's no longer necessary. I've removed it.